test/test_corpus.cc

```diff
@@ -132,6 +132,56 @@ unique_ptr<Diagnostic> errorToDiagnostic(const core::GlobalState &gs, const core
     return make_unique<Diagnostic>(Range::fromLoc(gs, error.loc), error.header);
 }
 
+class ErrorHandler {
+    vector<unique_ptr<core::Error>> &errors;
+    Expectations &test;
+    shared_ptr<core::ErrorQueue> errorQueue;
+
+public:
+    map<string, string> got;
+
+    ErrorHandler(vector<unique_ptr<core::Error>> &errors, Expectations &test, shared_ptr<core::ErrorQueue> errorQueue)
+        : errors(errors), test(test), errorQueue(errorQueue){};
+
+    void checkExpectations(string expectationType, std::function<string()> mkExp, bool addNewline = true) {
```

Comment:
Nit: Can `expectationType` be a `string_view` (or does that not work with `got.contains`)? I believe it should work with the `UnorderedMap`...

Reply:
Also, `checkExpectations` is a misnomer; this is actually more like `storeActual`?